src/OrchardCore.Modules/OrchardCore.HealthChecks/Startup.cs

```diff
@@ -17,17 +21,38 @@ public Startup(IShellConfiguration shellConfiguration)
             _shellConfiguration = shellConfiguration;
         }
 
-        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        public override void ConfigureServices(IServiceCollection services)
         {
-            var healthCheckOptions = serviceProvider.GetService<IOptions<HealthChecksOptions>>().Value;
+            services.AddScoped<IHealthChecksResponseWriter, DefaultHealthChecksResponseWriter>();
+            services.AddHealthChecks();
 
-            routes.MapHealthChecks(healthCheckOptions.Url);
+            services.Configure<HealthChecksOptions>(_shellConfiguration.GetSection("OrchardCore_HealthChecks"));
         }
 
-        public override void ConfigureServices(IServiceCollection services)
+        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
         {
-            services.AddHealthChecks();
-            services.Configure<HealthChecksOptions>(_shellConfiguration.GetSection("OrchardCore_HealthChecks"));
+            var healthChecksOptions = serviceProvider.GetService<IOptions<HealthChecksOptions>>().Value;
+
+            if (healthChecksOptions.ShowDetails)
+            {
+                var healthChecksResponseWriter = serviceProvider.GetService<IHealthChecksResponseWriter>();
+
+                routes.MapHealthChecks(healthChecksOptions.Url, new HealthCheckOptions
```

Comment:
Same comment as https://github.com/OrchardCMS/OrchardCore/pull/13592#discussion_r1174454987

It does not do exactly the same, `MapHealthChecks()` will build a new app builder (we have one here) to create a new pipeline (will shortcut the tenant pipeline), so I think that still using `UseHealthChecks()` is better.